src/main/java/SYSC4806/Controller/HomeController.java

```diff
@@ -40,27 +43,73 @@ public String home(Model model) {
     }
 
     /*
-    * Selects courses based on chosen program and year and returns them to view
-    * to refresh results table
+     * Selects courses based on chosen program, year, courses and catagories and returns them to view
+     * to refresh results table
      */
     @GetMapping("/results/{program}/{year}")
-    public String showResults(Model model, @PathVariable("program") String p, @PathVariable String year, @RequestParam(value = "courses", required = false) String co)
-    {
+    public String showResults(Model model, @PathVariable("program") String p, @PathVariable String year,
+                              @RequestParam(value = "courses", required = false) String co,
+                              @RequestParam(value = "categories", required = false) String ca) {
+
         long p_id = programRepository.findByName(p).getId();
         List<String> courseNames = courseRepository.findCourseByProgramAndYear(p_id, Integer.parseInt(year));
 
-        //Edge case: filter doesnt come up with something
-        if(co!=null){
+        if (co != null && !(co.isEmpty())) {
             List<String> courseNamesFilter = Arrays.asList(co.split("\\s*,\\s*"));
```

Comment:
Should we consider changing this regex just to the comma character? I guess I don't really understand why we're performing two greedy searches on white space before and after the comma. In general, greedy searches are very inefficient and we are providing the format of the course offerings so we know exactly how they are formatted.

Reply:
Just clarifying this for you, in Java you split on a Regex matching string. The `\\s*,\\s*` means it's greedy searching for a zero to many white spaces followed by a comma followed by another search for zero to many white spaces.

I was just saying, the process time for greedy searches is costly and we could probably just do something like `co.split(",")` because we know the exact formatting of the string.

Reply:
Oh sorry! I was thinking you were talking about this (co != null && !(co.isEmpty())) and got confused. I get it yup!